modules/rostests/apitests/kernel32/InitOnce.c

```diff
@@ -0,0 +1,280 @@
+     * initialization count should be 1 and retrieve correct context data.
+     */
+    ulInitCount = 0;
+    ulContextData = MAXULONG;
+    bRet = pfnInitOnceExecuteOnce(&InitOnce, InitOnceProc, &ulInitCount, (LPVOID*)&ulContextData);
+    ok(bRet, "InitOnceExecuteOnce failed with %lu\n", GetLastError());
+    if (bRet)
+    {
+        /* Call InitOnceExecuteOnce again and check output values if the first call succeeded */
+        bRet = pfnInitOnceExecuteOnce(&InitOnce,
+                                      InitOnceProc,
+                                      &ulInitCount,
+                                      (LPVOID*)&ulContextData);
+        ok(bRet, "InitOnceExecuteOnce failed with %lu\n", GetLastError());
+        ok(ulInitCount == 1, "ulInitCount is not 1\n");
+        ok(ulContextData == g_ulRandom, "Output ulContextData is incorrect\n");
+    }
+
+_test_sync:
+    if (!pfnInitOnceBeginInitialize || !pfnInitOnceComplete)
+    {
+        skip("InitOnceBeginInitialize or InitOnceComplete not found\n");
+        goto _exit;
+    }
+
+    /* Re-initialize One-Time initialization structure by using INIT_ONCE_STATIC_INIT */
+    InitOnce = (INIT_ONCE)INIT_ONCE_STATIC_INIT;
+
+    /* Perform synchronous initialization by using InitOnceBeginInitialize */
+    fPending = FALSE;
+    bRet = pfnInitOnceBeginInitialize(&InitOnce, 0, &fPending, (LPVOID*)&ulContextData);
+    ok(bRet, "InitOnceBeginInitialize failed with %lu\n", GetLastError());
+    if (!bRet)
+    {
+        goto _test_async;
+    }
+    ok(fPending, "fPending is not TRUE after the first success InitOnceBeginInitialize\n");
+    if (!fPending)
+    {
+        goto _test_async;
+    }
+
+    /* Complete the initialization */
+    InitWorker(&ulInitCount, &ulTempContext);
+    bRet = pfnInitOnceComplete(&InitOnce, 0, (LPVOID)ulTempContext);
+    ok(bRet, "InitOnceComplete failed with %lu\n", GetLastError());
+    if (!bRet)
+    {
+        goto _test_async;
+    }
+
+    /*
+     * Initialization is completed, call InitOnceBeginInitialize with
+     * INIT_ONCE_CHECK_ONLY should retrieve status and context data successfully
+     */
+    bRet = pfnInitOnceBeginInitialize(&InitOnce,
+                                      INIT_ONCE_CHECK_ONLY,
+                                      &fPending,
+                                      (LPVOID*)&ulContextData);
+    ok(bRet && !fPending && ulContextData == g_ulRandom,
+       "InitOnceBeginInitialize returns incorrect result for a completed initialization\n");
+
+_test_async:
+    InitOnce = (INIT_ONCE)INIT_ONCE_STATIC_INIT;
+
+    /* Perform asynchronous initialization */
+    fPending = FALSE;
+    bRet = pfnInitOnceBeginInitialize(&InitOnce,
+                                      INIT_ONCE_ASYNC,
+                                      &fPending,
+                                      (LPVOID*)&ulContextData);
+    ok(bRet, "InitOnceBeginInitialize failed with %lu\n", GetLastError());
+    if (!bRet)
+    {
+        goto _exit;
```

Comment:
Why not just doing `return;` here? Unless you expect to have in the future some extra cleanup going on after the `_exit` label?

Reply:
Committed. And also changed `goto _exit;` to `return;` in other places.